sources/Parser/ParserYaml.cpp

```diff
@@ -0,0 +1,381 @@
+	{
+		std::string tmpName = "";
+		Vector3f pos;
+		MapGenerator::BlockType blockType = MapGenerator::NOTHING;
+
+		for (; index != lines.size(); index++) {
+			if (lines[index].find("position") != std::string::npos) {
+				pos = _parsePosition(lines[index]);
+			} else if (lines[index].find("block_type") != std::string::npos) {
+				blockType = _parseBlockType(lines[index]);
+			} else {
+				if (!tmpName.empty()) {
+					break;
+				}
+				tmpName = lines[index];
+			}
+		}
+		if (blockType == MapGenerator::NOTHING)
+			throw (ParserError("Invalid block_type field."));
+		if (blockType == MapGenerator::HOLE)
+			pos.y += 1.0f;
+		map[std::make_tuple(pos.x, pos.y, pos.z)] = blockType;
+	}
+
+	void ParserYAML::_loadBlocks(std::shared_ptr<WAL::Scene> scene)
+	{
+		std::ifstream file("save/" + fileName + "_block.yml");
+		std::string line;
+		std::vector<std::string> lines;
+		MapGenerator::MapBlock map;
+
+		if (!file.good())
+			throw (ParserError("File error"));
+		for (int i = 0; i < Runner::width; i++)
+			for (int j = 0; j < Runner::height; j++)
+				map[std::make_tuple(i, 0, j)] = MapGenerator::NOTHING;
+		while (std::getline(file, line)) {
+			if (line.find("width:") != std::string::npos) {
+				Runner::width = std::atoi(line.substr(line.find(": ", 0) + 2, line.length()).c_str());
+				continue;
+			}
+			if (line.find("height:") != std::string::npos) {
+				Runner::height = std::atoi(line.substr(line.find(": ", 0) + 2, line.length()).c_str());
+				continue;
+			}
+			if (line.empty() || line.find("blocks:") != std::string::npos)
+				continue;
+			lines.push_back(line);
+		}
+		for (int index = 0; lines.size() != index; index++) {
+			_loadBlock(scene, lines, index, map);
+			index--;
+		}
+		MapGenerator::loadMap(Runner::width, Runner::height, map, scene);
+	}
+
+	void ParserYAML::_loadBonus(std::shared_ptr<WAL::Scene> scene, std::vector<std::string> lines, int &index)
+	{
+		static std::map<Bonus::BonusType, std::vector<std::string>> map = {
+				{Bonus::BonusType::BOMBSTOCK, {"Bonus Bomb Up", "assets/items/bombup"}},
+				{Bonus::BonusType::SPEEDUP, {"Bonus Speed Up", "assets/items/speedup"}},
+				{Bonus::BonusType::EXPLOSIONINC, {"Bonus Fire Up", "assets/items/fireup"}}
+		};
+		static std::vector<std::function<void (WAL::Entity &, const WAL::Entity &, CollisionComponent::CollidedAxis)>> func = {
+				&Bonus::BombUpBonus, &Bonus::SpeedUpBonus, &Bonus::ExplosionRangeBonus
+		};
+		auto &entity = scene->addEntity("");
+		std::string tmpName = "";
+		Vector3f pos;
+		Bonus::BonusType bonusType;
+
+		for (; index != lines.size(); index++) {
+			if (lines[index].find("position") != std::string::npos) {
+				pos = _parsePosition(lines[index]);
+			} else if (lines[index].find("bonus_type") != std::string::npos) {
+				bonusType = _parseBonusType(lines[index]);
+			} else {
+				if (!tmpName.empty()) {
+					break;
+				}
+				_parseEntityName(lines[index], entity);
+			}
+		}
+		entity.addComponent<PositionComponent>(pos)
```

Comment:
you should create a function createBonus that will be common here and in Map.cpp